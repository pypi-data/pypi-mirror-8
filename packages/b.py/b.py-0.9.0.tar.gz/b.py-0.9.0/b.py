#!/usr/bin/env python
# Copyright (C) 2013 by Yu-Jie Lin
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in
# all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
# THE SOFTWARE.

"""
============
b.py command
============


Commands
========

============= =======================
command       supported services
============= =======================
``blogs``     ``b``
``post``      ``b``, ``wp``
``generate``  ``base``, ``b``, ``wp``
``checklink`` ``base``, ``b``, ``wp``
``search``    ``b``
============= =======================

Descriptions:

``blogs``
  list blogs. This can be used for blog IDs lookup.

``post``
   post or update a blog post.

``generate``
  generate HTML file at ``<TEMP>/draft.html``, where ``<TEMP>`` is the system's
  temporary directory.

  The generation can output a preview html at ``<TEMP>/preview.html`` if there
  is ``tmpl.html``. It will replace ``%%Title%%`` with post title and
  ``%%Content%%`` with generated HTML.

``checklink``
  check links in generated HTML using lnkckr_.

``search``
  search blog

.. _lnkckr: https://pypi.python.org/pypi/lnkckr
"""

from __future__ import print_function

import argparse as ap
import imp
import os
import sys
import traceback

from bpy.handlers import handlers
from bpy.services import find_service, services

__program__ = 'b.py'
__description__ = 'Post to Blogger or WordPress in markup language seamlessly'
__copyright__ = 'Copyright 2013-2014, Yu Jie Lin'
__license__ = 'MIT License'
__version__ = '0.9.0'
__website__ = 'http://bitbucket.org/livibetter/b.py'

__author__ = 'Yu-Jie Lin'
__author_email__ = 'livibetter@gmail.com'


# b.py stuff
############

# filename of local configuration without '.py' suffix.
BRC = 'brc'


def parse_args():

  p = ap.ArgumentParser()
  p.add_argument('--version', action='version',
                 version='%(prog)s ' + __version__)
  p.add_argument('-s', '--service', default='base',
                 help='what service to use. (Default: %(default)s)')
  sp = p.add_subparsers(help='commands')

  pblogs = sp.add_parser('blogs', help='list blogs')
  pblogs.set_defaults(subparser=pblogs, command='blogs')

  psearch = sp.add_parser('search', help='search for posts')
  psearch.add_argument('-b', '--blog', help='Blog ID')
  psearch.add_argument('q', nargs='+', help='query text')
  psearch.set_defaults(subparser=psearch, command='search')

  pgen = sp.add_parser('generate', help='generate html')
  pgen.add_argument('filename')
  pgen.set_defaults(subparser=pgen, command='generate')

  pchk = sp.add_parser('checklink', help='check links in chkerateed html')
  pchk.add_argument('filename')
  pchk.set_defaults(subparser=pchk, command='checklink')

  ppost = sp.add_parser('post', help='post or update a blog post')
  ppost.add_argument('filename')
  ppost.set_defaults(subparser=ppost, command='post')

  args = p.parse_args()
  return args


def load_config():

  rc = None
  try:
    search_path = [os.getcwd()]
    _mod_data = imp.find_module(BRC, search_path)
    print('Loading local configuration...')
    try:
      rc = imp.load_module(BRC, *_mod_data)
    finally:
      if _mod_data[0]:
        _mod_data[0].close()
  except ImportError:
    pass
  except Exception:
    traceback.print_exc()
    print('Error in %s, aborted.' % _mod_data[1])
    sys.exit(1)
  return rc


def main():

  args = parse_args()

  rc = load_config()
  service_options = {'blog': None}
  if rc:
    if hasattr(rc, 'handlers'):
      for name, handler in rc.handlers.items():
        if name in handlers:
          handlers[name].update(handler)
        else:
          handlers[name] = handler.copy()
    if hasattr(rc, 'services'):
      for name, service in rc.services.items():
        if name in services:
          services[name].update(service)
        else:
          services[name] = service.copy()
    if hasattr(rc, 'service'):
      args.service = rc.service
    if hasattr(rc, 'service_options'):
      service_options.update(rc.service_options)

  if hasattr(args, 'blog') and args.blog is not None:
    service_options['blog'] = args.blog
  filename = args.filename if hasattr(args, 'filename') else None
  service = find_service(args.service, service_options, filename)

  if args.command == 'blogs':
    service.list_blogs()
  elif args.command == 'search':
    service.search(' '.join(args.q))
  elif args.command == 'generate':
    service.generate()
  elif args.command == 'checklink':
    service.checklink()
  elif args.command == 'post':
    service.post()


if __name__ == '__main__':
  main()
