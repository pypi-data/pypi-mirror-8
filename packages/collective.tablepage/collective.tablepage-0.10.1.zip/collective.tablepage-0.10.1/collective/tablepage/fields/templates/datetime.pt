<tal:calendar define="formname string:edit-row;
                      inputname view/configuration/id;
                      inputvalue options/data|nothing;
                      show_hm view/show_hm;
                      uniqueItemIndex view/RealIndexIterator;
                      here nocall:context;
                      ">
<label class="subFieldLabel"
       tal:attributes="for view/configuration/id"
       tal:content="view/configuration/label">A good field</label>
<span class="fieldRequired" title="Required"
      tal:condition="view/options/required"
      i18n:domain="plone"
      i18n:attributes="title title_required;"
      i18n:translate="label_required">(Required)</span>
<div class="formHelp" tal:content="view/configuration/description">
    Help filling me
</div>
<metal:box use-macro="context/calendar_macros/macros/calendarDatePickerBox">
  <!-- a calendar, hopefully -->
</metal:box>
</tal:calendar>