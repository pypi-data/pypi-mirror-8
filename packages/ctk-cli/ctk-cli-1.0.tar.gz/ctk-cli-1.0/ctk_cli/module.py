# see https://github.com/commontk/CTK/blob/master/Libs/CommandLineModules/Core/Resources/ctkCmdLineModule.xsd
# for what we aim to be able to parse

import os, logging
import xml.etree.ElementTree as ET
from execution import isCLIExecutable, getXMLDescription

logger = logging.getLogger(__name__)


# --------------------------------------------------------------------

def _tagToIdentifier(tagName):
    return tagName.replace('-', '_')

def _parseBool(x):
    if x in ('true', 'True', '1'):
        return True
    if x in ('false', 'False', '0'):
        return False
    raise ValueError, "cannot convert %r to boolean" % (x, )

def _tag(element):
    """Return element.tag with xmlns stripped away."""
    tag = element.tag
    if tag[0] == "{":
        uri, tag = tag[1:].split("}")
    return tag

def _uriPrefix(element):
    """Return xmlns prefix of the given element."""
    i = element.tag.find('}')
    if i < 0:
        return ""
    return element.tag[:i+1]


def _parseElements(self, elementTree, expectedTag = None):
    """Read REQUIRED_ELEMENTS and OPTIONAL_ELEMENTS and returns
    the rest of the children.  Every read child element's text
    value will be filled into an attribute of the same name,
    i.e. <description>Test</description> will lead to 'Test' being
    assigned to self.description.  Missing REQUIRED_ELEMENTS
    result in warnings."""

    xmlns = _uriPrefix(elementTree)

    if expectedTag is not None:
        assert _tag(elementTree) == expectedTag, 'expected <%s>, got <%s>' % (expectedTag, _tag(elementTree))

    parsed = []

    for tagName in self.REQUIRED_ELEMENTS + self.OPTIONAL_ELEMENTS:
        tags = elementTree.findall(xmlns + tagName)
        if tags:
            parsed.extend(tags)
            tagValue = tags[0].text
            tagValue = tagValue.strip() if tagValue else ""
            if len(tags) > 1:
                logger.warning("More than one <%s> found within %r (using only first)" % (tagName, _tag(elementTree)))
        else:
            tagValue = None
            if tagName in self.REQUIRED_ELEMENTS:
                logger.warning("Required element %r not found within %r" % (tagName, _tag(elementTree)))
        setattr(self, _tagToIdentifier(tagName), tagValue)

    return [tag for tag in elementTree if tag not in parsed]


class CLIModule(list):
    REQUIRED_ELEMENTS = ('title', 'description')

    OPTIONAL_ELEMENTS = ('category', 'index', 'version', 'documentation-url',
                         'license', 'contributor', 'acknowledgements')

    __slots__ = ('path', ) + tuple(map(_tagToIdentifier, REQUIRED_ELEMENTS + OPTIONAL_ELEMENTS))

    def __init__(self, path, env = None):
        self.path = path

        if isCLIExecutable(path):
            elementTree = getXMLDescription(path, env = env)
        else:
            with file(path) as f:
                elementTree = ET.parse(f)

        self._parse(elementTree.getroot())

    @property
    def name(self):
        result = os.path.basename(self.path)
        if result.endswith('.exe'):
            result = result[:-4]
        return result

    def parameters(self):
        """Generator that recursively enumerates all parameters from
        all parameter groups"""
        for parameters in self:
            for parameter in parameters:
                yield parameter

    def classifyParameters(self):
        """Return (arguments, options, outputs) tuple.  Together, the
        three lists contain all parameters (recursively fetched from
        all parameter groups), classified into optional parameters,
        required ones (with an index), and simple output parameters
        (that would get written to a file using
        --returnparameterfile).  `arguments` contains the required
        arguments, already sorted by index."""

        arguments = []
        options = []
        outputs = []
        for parameter in self.parameters():
            if parameter.channel == 'output' and not (
                    parameter.isExternalType() or parameter.typ == 'file'):
                outputs.append(parameter)
            elif parameter.index is not None:
                arguments.append(parameter)
                if parameter.flag is not None or parameter.longflag is not None:
                    logger.warning("Parameter %s has both index=%d and flag set." % (
                        parameter.identifier(), parameter.index))
            else:
                options.append(parameter)
        arguments.sort(key = lambda parameter: parameter.index)
        return (arguments, options, outputs)

    # this is called _parse, not parse (like in the classes below),
    # because it is not a classmethod that is supposed to be used as a
    # factory method from the outside, even if the signature and
    # content is really similar:
    def _parse(self, elementTree):
        childNodes = _parseElements(self, elementTree, 'executable')

        for pnode in childNodes:
            if _tag(pnode) == 'parameters':
                self.append(CLIParameters.parse(pnode))
            else:
                logger.warning("Element %r within %r not parsed" % (_tag(pnode), _tag(elementTree)))


class CLIParameters(list):
    REQUIRED_ELEMENTS = ('label', 'description')
    OPTIONAL_ELEMENTS = ()

    __slots__ = ("advanced", ) + REQUIRED_ELEMENTS

    @classmethod
    def parse(cls, elementTree):
        self = cls()
        
        childNodes = _parseElements(self, elementTree, 'parameters')
        
        self.advanced = _parseBool(elementTree.get('advanced', 'false'))

        for pnode in childNodes:
            self.append(CLIParameter.parse(pnode))

        return self


class CLIParameter(object):
    VALUE_TYPES = (
        'boolean',
        'integer', 'float', 'double',
        'string', 'directory',
        'integer-vector', 'float-vector', 'double-vector',
        'string-vector',
        'integer-enumeration', 'float-enumeration', 'double-enumeration', 'string-enumeration',
        'point', 'region',
        'file',
    )

    # parameter types which are passed through temporary files (with default extensions)
    EXTERNAL_TYPES = {
        'image'       : '.nrrd',
        'geometry'    : '.vtp',
        'transform'   : '.mrml',
        'table'       : '.ctbl', # (CSV format)
        'measurement' : '.csv',
    }

    TYPES = VALUE_TYPES + tuple(EXTERNAL_TYPES)

    PYTHON_TYPE_MAPPING = {
        'boolean' : bool,
        'integer' : int,
        'float' : float,
        'double' : float,
    }

    REQUIRED_ELEMENTS = ('name', 'description', 'label')

    OPTIONAL_ELEMENTS = (# either 'flag' or 'longflag' (or both) or 'index' are required
                         'flag', 'longflag', 'index',
                         'default', 'channel')
    
    __slots__ = ("typ", "hidden", "_pythonType") + REQUIRED_ELEMENTS + OPTIONAL_ELEMENTS + (
                 "constraints", # scalarVectorType, scalarType
                 "multiple", # multipleType
                 "elements", # enumerationType
                 "coordinateSystem", # pointType
                 "fileExtensions", # fileType
                 "subtype", # 'type' of imageType / geometryType
        )

    def identifier(self):
        result = self.name if self.name else self.longflag.lstrip('-')
        if not result:
            raise RuntimeError, "Cannot identify parameter either by name or by longflag (both missing)"
        return result

    def parseValue(self, value):
        """Parse the given value and return result."""
        if self.isNumericVector():
            return map(self._pythonType, value.split(','))
        if self.typ == 'boolean':
            return _parseBool(value)
        return self._pythonType(value)

    def isOptional(self):
        return self.index is None

    def isNumericVector(self):
        """Return whether this is a vector-like type with multiple
        elements of a numeric type.  This includes the numeric
        xxx-vector types as well as point (fixed 3D) and region (fixed
        6D).  For these, the value will be a sequence of the
        corresponding python type."""
        return (self.typ.endswith('-vector') and self.typ != 'string-vector') or self.typ in ('point', 'region')

    def isExternalType(self):
        """Return True iff parameter values of this type are transferred via (temporary) files"""
        return self.typ in self.EXTERNAL_TYPES

    def defaultExtension(self):
        """Return default extension for this parameter type, checked
        against supported fileExtensions.  If the default extension is
        not within `fileExtensions`, return the first supported
        extension."""
        result = self.EXTERNAL_TYPES[self.typ]
        if not self.fileExtensions:
            return result
        if result in self.fileExtensions:
            return result
        return self.fileExtensions[0]

    @classmethod
    def parse(cls, elementTree):
        assert _tag(elementTree) in cls.TYPES, _tag(elementTree)

        self = cls()
        self.typ = _tag(elementTree)

        if self.typ in ('point', 'region'):
            self._pythonType = float
        else:
            elementType = self.typ
            if elementType.endswith('-vector'):
                elementType = elementType[:-7]
            elif elementType.endswith('-enumeration'):
                elementType = elementType[:-12]
            self._pythonType = self.PYTHON_TYPE_MAPPING.get(elementType, str)

        self.hidden = _parseBool(elementTree.get('hidden', 'false'))

        self.constraints = None
        self.multiple = None
        self.elements = None
        self.coordinateSystem = None
        self.fileExtensions = None
        self.subtype = None

        for key, value in elementTree.items():
            if key == 'multiple':
                self.multiple = _parseBool(value)
            elif key == 'coordinateSystem':
                self.coordinateSystem = value
            elif key == 'coordinateSystem':
                self.coordinateSystem = value
            elif key == 'fileExtensions':
                self.fileExtensions = [ext.strip() for ext in value.split(",")]
            elif key == 'type':
                self.subtype = value
            elif key != 'hidden':
                logger.warning('attribute of %r ignored: %s=%r' % (_tag(elementTree), key, value))

        elements = []

        childNodes = _parseElements(self, elementTree)
        for n in childNodes:
            if _tag(n) == 'constraints':
                self.constraints = CLIConstraints.parse(n)
            elif _tag(n) == 'element':
                if not n.text:
                    logger.warning("Ignoring empty <element> within <%s>" % (_tag(elementTree), ))
                else:
                    elements.append(n.text)
            else:
                logger.warning("Element %r within %r not parsed" % (_tag(n), _tag(elementTree)))

        if self.flag and not self.flag.startswith('-'):
            self.flag = '-' + self.flag
        if self.longflag and not self.longflag.startswith('-'):
            self.longflag = '--' + self.longflag

        if self.index is not None:
            self.index = int(self.index)

        if self.default:
            self.default = self.parseValue(self.default)

        if self.typ.endswith('-enumeration'):
            self.elements = map(self.parseValue, elements)
            if not elements:
                logger.warning("No <element>s found within <%s>" % (_tag(elementTree), ))
        else:
            self.elements = None
            if elements:
                logger.warning("Ignoring <element>s within <%s>" % (_tag(elementTree), ))

        return self


class CLIConstraints(object):
    REQUIRED_ELEMENTS = ('step', )

    OPTIONAL_ELEMENTS = ('minimum', 'maximum')
    
    __slots__ = REQUIRED_ELEMENTS + OPTIONAL_ELEMENTS

    @classmethod
    def parse(cls, elementTree):
        self = cls()
        childNodes = _parseElements(self, elementTree, 'constraints')
        for n in childNodes:
            logger.warning("Element %r within %r not parsed" % (_tag(n), _tag(elementTree)))

        return self
