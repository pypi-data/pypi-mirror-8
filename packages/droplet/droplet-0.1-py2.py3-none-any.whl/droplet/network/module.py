from droplet import module
from droplet.files import TemplateConfFile
from droplet.commands import run
from droplet.sudo import root

from .models import Interface

import logging


logger = logging.getLogger(__name__)


class Network(module.Module):

    # Configuration files

    interfaces = TemplateConfFile('/etc/network/interfaces',
                                  template='interfaces',
                                  template_params=lambda: {
                                      'interfaces': Interface.objects.all()
                                  })

    def start(self):
        for iface in Interface.objects.all():
            if not iface.configured:
                logger.debug("%s is not configured" % iface.name)
                continue

            with root():
                # we run ifup in background
                # it can wait indefinitely for DHCP requests
                run('/sbin/ifup --force %s' % iface.name, background=True)

    def stop(self):
        for iface in Interface.objects.all():
            if not iface.configured:
                logger.debug("%s is not configured" % iface.name)
                continue

            with root():
                run('/sbin/ifdown --force %s' % iface.name)
