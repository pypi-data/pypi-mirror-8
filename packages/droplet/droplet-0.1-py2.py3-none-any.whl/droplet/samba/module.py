from droplet import module
from droplet.files import TemplateConfFile
from droplet.daemons import InitDaemon
from droplet.commands import run
from droplet.sudo import root
from droplet.interfaces.samba import ISambaServer
from droplet.util import make_password

import os
import shutil
import logging

from .models import DomainSettings


logger = logging.getLogger(__name__)


class Samba(module.Module, ISambaServer):
    """
    Samba 4 module, it deploys samba AD and file server
    """
    install_wizard = 'samba:install'

    SMBCONF_FILE = '/etc/samba/smb.conf'
    SMB_PRIVATE_DIR = '/var/lib/samba/private'
    KRB5CONF_FILE = '/etc/krb5.conf'
    SMB_KRB5CONF_FILE = os.path.join(SMB_PRIVATE_DIR, 'krb5.conf')

    # Conf files
    smbconf = TemplateConfFile(SMBCONF_FILE,
                               template='smb.conf',
                               template_params=lambda: {
                                   'settings': DomainSettings.get()
                               })

    # Daemons
    nmbd = InitDaemon('nmbd')
    smbd = InitDaemon('smbd')
    samba_ad = InitDaemon('samba-ad-dc')

    def install(self):
        """
        Installation procedure, it writes basic smb.conf and uses samba-tool to
        provision the domain
        """
        domain_settings = DomainSettings.get()

        with root():
            if os.path.exists(self.SMBCONF_FILE):
                os.remove(self.SMBCONF_FILE)

            if domain_settings.mode == 'ad':
                domain_settings.adminpass = make_password(15)
                domain_settings.save()

                run("samba-tool domain provision "
                    "--domain='%s' "
                    "--workgroup='%s' "
                    "--realm='%s' "
                    "--use-xattrs=yes "
                    "--use-rfc2307 "
                    "--server-role='domain controller' "
                    "--use-ntvfs "
                    "--adminpass='%s'" %
                    (domain_settings.domain,
                     domain_settings.workgroup,
                     domain_settings.realm,
                     domain_settings.adminpass))

                self.smbconf.write()

                shutil.copy2(self.SMB_KRB5CONF_FILE, self.KRB5CONF_FILE)

                # XXX FIXME move this to network
                run("echo 'nameserver 127.0.0.1' > /etc/resolv.conf")
                # TODO manage shares
                run("touch /etc/samba/shares.conf")

            elif domain_settings.mode == 'member':
                # TODO
                pass

    def save(self):

        # Save con files
        self.smbconf.write()

        with root():
            shutil.copy2(self.SMB_KRB5CONF_FILE, self.KRB5CONF_FILE)
            # XXX FIXME move this to network
            run("echo 'nameserver 127.0.0.1' > /etc/resolv.conf")

    def stop_other_daemons(self):
        """
        Stop services already provided by main samba daemon
        """
        if self.smbd.running:
            self.smbd.stop()

        if self.nmbd.running:
            self.nmbd.stop()

    def start(self):
        self.stop_other_daemons()
        self.samba_ad.start()

    def stop(self):
        self.stop_other_daemons()
        self.samba_ad.stop()

    def restart(self):
        self.stop_other_daemons()
        self.samba_ad.restart()

    def connection_info(self):
        domain_settings = DomainSettings.get()
        base_dn = 'dc=' + ',dc='.join(domain_settings.realm.split('.'))

        return {
            'host': '127.0.0.1',
            'bind_dn': 'cn=Administrator,cn=Users,%s' % base_dn,
            'bind_password': domain_settings.adminpass,
            'base_dn': base_dn,
            'protocol': 'ldap'
        }
