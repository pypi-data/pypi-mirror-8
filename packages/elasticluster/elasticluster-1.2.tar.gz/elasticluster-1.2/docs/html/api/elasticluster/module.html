

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>elasticluster.module &mdash; elasticluster 1.0 documentation</title>
    
    <link rel="stylesheet" href="../../_static/default.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <link rel="top" title="elasticluster 1.0 documentation" href="../../index.html" />
    <link rel="up" title="Elasitcluster programming API" href="../index.html" />
    <link rel="next" title="elasticluster.providers" href="providers.html" />
    <link rel="prev" title="elasticluster.main" href="main.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="../../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="providers.html" title="elasticluster.providers"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="main.html" title="elasticluster.main"
             accesskey="P">previous</a> |</li>
        <li><a href="../../index.html">elasticluster 1.0 documentation</a> &raquo;</li>
          <li><a href="../index.html" accesskey="U">Elasitcluster programming API</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="module-elasticluster.module">
<span id="elasticluster-module"></span><h1><cite>elasticluster.module</cite><a class="headerlink" href="#module-elasticluster.module" title="Permalink to this headline">Â¶</a></h1>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="main.html"
                        title="previous chapter"><cite>elasticluster.main</cite></a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="providers.html"
                        title="next chapter"><cite>elasticluster.providers</cite></a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="../../_sources/api/elasticluster/module.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="../../py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="providers.html" title="elasticluster.providers"
             >next</a> |</li>
        <li class="right" >
          <a href="main.html" title="elasticluster.main"
             >previous</a> |</li>
        <li><a href="../../index.html">elasticluster 1.0 documentation</a> &raquo;</li>
          <li><a href="../index.html" >Elasitcluster programming API</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Grid Computing Competence Centre, University of Zurich.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>