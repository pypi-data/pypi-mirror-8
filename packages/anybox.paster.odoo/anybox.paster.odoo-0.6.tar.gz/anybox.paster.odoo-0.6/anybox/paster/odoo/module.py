from paste.script.templates import Template
import datetime
import os
import ConfigParser
import re


def get_conf(conffile, section, option):
    author = ""
    author_email = ""
    val = ""
    p = ConfigParser.RawConfigParser()
    try:
        p.read([conffile])
        for o, v in p.items(section):
            if o == option:
                val = v
        if val:
            regex = re.compile('(.*) <(.*)>')
            author, author_email = regex.findall(val)[0]
    except Exception, e:
        print e

    return author, author_email


def git_conf(conffile, section, option):
    for line in open(conffile).readlines():
        if line.strip() == '[%s]' % section:
            user_section = True
        if user_section and line.split('=', 1)[0].strip() == option:
            return line.split('=', 1)[1].strip()


class Module(Template):
    summary = 'Template for creating an Odoo python file'
    required_templates = []
    _template_dir = 'templates/module'
    use_cheetah = True

    def pre(self, command, output_dir, vars):
        gitfile = os.path.expanduser("~/.gitconfig")
        hgfile = os.path.expanduser("~/.hgrc")
        bzrfile = os.path.expanduser("~/.bazaar/bazaar.conf")
        if os.path.exists(gitfile):
            author = git_conf(gitfile, 'user', 'name')
            author_email = git_conf(gitfile, 'user', 'email')
        elif os.path.exists(hgfile):
            author, author_email = get_conf(hgfile, 'ui', 'username')
        elif os.path.exists(bzrfile):
            author, author_email = get_conf(bzrfile, 'DEFAULT', 'email')

        loopvars = [
            ('description', 'Small description', ''),
            ('author', 'Author name', author),
            ('author_email', 'Author email', author_email),
            ('name', 'Module Name', vars['project'].replace('_', ' ')),
            ('version', 'Module version', '0.1'),
            ('category', 'Module category',  'Uncategorized'),
            ('company', 'Author Company', "Anybox"),
            ('website', 'Website', "http://anybox.fr"),
            ('depends', 'List of Odoo module dependencies (separated by ,)', "base"),
            ('app', 'Is the module an application? (True/False)', "False"),
        ]
        counter = 0
        vars['year'] = str(datetime.date.today().year)
        while counter < len(loopvars):
            field, question, default = loopvars[counter]
            if field in ('application',):
                question = command.challenge(question + ' :', default, True)
                if question not in ('True', 'False'):
                    continue
            else:
                question = command.challenge('Please Enter ' + question + ' :', default, True)
            if question != '':
                if field == 'depends':
                    question = ','.join(["'" + x.strip() + "'" for x in question.split(',')])
                vars[field] = question
                counter += 1
