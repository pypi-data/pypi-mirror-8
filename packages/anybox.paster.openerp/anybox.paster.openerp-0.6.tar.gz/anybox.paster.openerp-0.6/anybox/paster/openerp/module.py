from paste.script.templates import Template
import datetime
import os
import ConfigParser
import re


class Module(Template):
    egg_plugins = ['anybox_openerp']
    summary = 'Template for creating an OpenERP python file'
    required_templates = []
    _template_dir = 'templates/module'
    use_cheetah = True

    def pre(self, command, output_dir, vars):
        def get_conf(conffile, section, option):
            author = ""
            author_email = ""
            val = ""
            p = ConfigParser.ConfigParser()
            try:
                p.read([conffile])
                for o, v in p.items(section):
                    if o == option:
                        val = v
                if val:
                    regex = re.compile('(.*) <(.*)>')
                    author, author_email = regex.findall(val)[0]
            except Exception, e:
                print e

            return author, author_email

        hgfile = os.path.expanduser("~/.hgrc")
        bzrfile = os.path.expanduser("~/.bazaar/bazaar.conf")
        if os.path.exists(hgfile):
            author, author_email = get_conf(hgfile, 'ui', 'username')
        elif os.path.exists(bzrfile):
            author, author_email = get_conf(bzrfile, 'DEFAULT', 'email')

        loopvars = [
            ('description', 'Small description', ''),
            ('author', 'Author name', author),
            ('author_email', 'Author email', author_email),
            ('name', 'Module Name', vars['project'].replace('_', ' ')),
            ('version', 'Module version', '0.1'),
            ('category', 'Module category',  vars['project'].replace('_', ' ')),
            ('company', 'Author Company', "Anybox"),
            ('website', 'Website', "http://anybox.fr"),
            ('depends', 'List of OpenERP module dependencies (separated by ,)', "base"),
            ('app', 'Is the module an application? (True/False)', "False"),
            ('active', 'Should the module be installed automatically? (True/False)', "False"),
        ]
        counter = 0
        vars['year'] = str(datetime.date.today().year)
        while counter < len(loopvars):
            field, question, default = loopvars[counter]
            if field in ('active', 'application'):
                question=command.challenge(question + ' :', default, True)
                if question not in ('True', 'False'):
                    continue
            else:
                question=command.challenge('Please Enter ' + question + ' :', default, True)
            if question != '':
                if field == 'depends':
                    question =  ','.join(["'" + x.strip() + "'" for x in question.split(',')])
                vars[field] = question
                counter += 1
