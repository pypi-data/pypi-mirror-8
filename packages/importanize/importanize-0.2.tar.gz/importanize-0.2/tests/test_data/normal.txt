"""Docstring here"""

from __future__ import unicode_literals, print_function


import os.path as ospath
import datetime
from package.subpackage.module.submodule import CONSTANT, Klass, foo, bar, rainbows
from .module import foo, bar
from ..othermodule import rainbows
from a import b
from a.b import c
from a.b import d



# stuff here
