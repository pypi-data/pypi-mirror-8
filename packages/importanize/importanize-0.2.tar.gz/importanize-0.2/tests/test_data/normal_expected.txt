"""Docstring here"""

from __future__ import print_function, unicode_literals
import datetime
from os import path as ospath

from a import b
from a.b import c, d
from package.subpackage.module.submodule import (
    CONSTANT,
    Klass,
    bar,
    foo,
    rainbows,
)

from ..othermodule import rainbows
from .module import bar, foo


# stuff here
